src/test/e2e/helpers/test-helper.ts

```diff
@@ -18,6 +18,53 @@ export interface IUnleashTest {
     destroy: () => Promise<void>;
     services: IUnleashServices;
     config: IUnleashConfig;
+    api: IUnleashHttpAPI;
+}
+
+export interface IUnleashHttpAPI {
+    createFeature(name: string, project?: string): supertest.Test;
+    archiveFeature(name: string, project?: string): supertest.Test;
+    createContextField(contextField: IContextFieldDto): supertest.Test;
+}
+
+export class UnleashHttpAPI implements IUnleashHttpAPI {
+    private request: supertest.SuperAgentTest;
+
+    private base: string;
+
+    constructor(request: supertest.SuperAgentTest, config: IUnleashConfig) {
+        this.request = request;
+        this.base = config.server.baseUriPath || '';
+    }
+
+    createFeature(
```

Comment:
I believe it's better to have the api field as the entry field, because that way we can do `api.` and autocomplete. Making it a function and passing the app as parameter instead of calling the function from the app, doesn't add a lot of value (at least for testing). 

With regards to the expected value, I think it's good that we expose Supertest, there might be situations where we want to assert different things on the response or headers. Hiding supertest means we'll probably need to proxy a lot of this functionality, and even though this could make sense for production code I don't think it does make sense for test where we want to be more flexible and err to development speed.

Reply:
+1 to exposing supertest for composability. I'd rather not have another abstraction on top

Reply:
Sticking all functions on an object is fine if you want autocompletion. I'd just use a regular object instead of a class that creates this object

Reply:
That's exactly what I meant :)